ui: [depends on #42820] Collapsible panel for Decommissioned nodes list

koorosh opened the pull request:
It should be possible to hide the Decommissioned nodes list because for some users it is not important information at all. To accomplish this, the `TableSection` component has been implemented where main purpose is to display the title of the section, content, and a footer and provide the ability to hide content from the user.
The `TableSection` component requires `id` property which is used to construct a unique key for LocalSettings storage.

In addition to the main changes, some styles for the `NodesList` component have been adjusted to reflect the latest designs. Particularly spacing was added between sections in the `NodesList` component.

koorosh committed:
ui: Add route to nodes history page

It has to be discussed about appropriate path name.

Release note: None

koorosh committed:
ui: Add Decommissioned nodes history page

- url path for new page is following: '#/reports/nodes/history',
please let me know if it has to be changed to something more meaningful.
I decided to make it more generic in case we want to extend it with
other 'historical' content. Also, it is nested route under 'nodes' as it
belongs to this entity.
- this page is supposed to display the history of all decommissioned
nodes in one place;
- the logic of retrieving and displaying nodes are almost the same as
in Nodes Overview container with adjustments according to provided
designs;
- Redux selectors, and helper functions are refactored to external
utils module so it reduces code duplication;

Release note: None

koorosh committed:
ui: Fix import paths in DecommissionedNodeHistory container

Release note: None

koorosh committed:
ui: Add 'Decommissioned node history' link on Nodes Overview page

Release note: None

koorosh committed:
ui: Add 'Decommissioned node history' link on Debug page

Release note: None

koorosh committed:
ui: Limit the number of most recent decommissioned nodes to display

- Display only 5 most recent decommissioned nodes on Cluster Overview view;
- Provide a link to navigate to Decommissioned Node History page;

Release note: None

koorosh committed:
ui: Display absolute time for decommissioned nodes

- Represent decommissioning date in absolute format instead of relative.

Release note: None

koorosh committed:
ui: Add TableSection reusable component

- TableSection component serves as a container which allows
collapse/expand its content

Release note: None

koorosh committed:
ui: Add ability to collapse Decommissioned nodes list on Nodes Overview page

- Reuse TableSection component to collapse/expand table
- Fix styles for Nodes List. Add spacing between nodes.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42958)
<!-- Reviewable:end -->

koorosh committed:
ui: Fix typo in a comment

Release note: None

koorosh committed:
ui: Remove saving isCollapsed state in componentWillUnmount hook in TableSection component

Extra saving isCollapsed state for TableSection component is unnecessary,
because in case a user does not change its state (it remains default value)
then we need to avoid unnecessary storing default values in session.

Release note: None

Koorosh commented:
Current changes are not valid anymore because of new UI changes on the Node List view.
The logic of collapsing/expanding table sections are moved to the branch: `ui-node-list-view-redesign`.
